dataRetriever/requestHandlers/requestHandler.go

```diff
@@ -115,6 +116,46 @@ func (rrh *resolverRequestHandler) RequestTransaction(destShardID uint32, txHash
 	rrh.requestByHashes(destShardID, txHashes, factory.TransactionTopic, uniqueTxSuffix)
 }
 
+// RequestReceipts method ask for receipts from the connected peers
+func (rrh *resolverRequestHandler) RequestReceipts(receiptsHashes [][]byte) {
+	unrequestedHashes := rrh.getUnrequestedHashes(receiptsHashes, uniqueReceiptSuffix)
+	if len(unrequestedHashes) == 0 {
+		return
+	}
+
+	log.Debug("requesting receipts from network",
+		"topic", factory.ReceiptTopic,
+		"num receipts", len(unrequestedHashes),
+	)
+
+	requester, err := rrh.requestersFinder.IntraShardRequester(factory.ReceiptTopic)
+	if err != nil {
+		log.Error("requestByHashes.CrossShardRequester",
+			"error", err.Error(),
+			"topic", factory.ReceiptTopic,
+		)
+		return
+	}
+
+	receiptRequester, ok := requester.(HashSliceRequester)
+	if !ok {
+		log.Warn("wrong assertion type when creating receipts requester")
+		return
+	}
+
+	for _, txHash := range receiptsHashes {
+		log.Trace("requestByHashes", "hash", txHash, "topic", factory.ReceiptTopic,
+			"num txs", len(unrequestedHashes),
+			"stack", string(debug.Stack()))
+	}
+
+	rrh.whiteList.Add(unrequestedHashes)
+
+	go rrh.requestHashesWithDataSplit(unrequestedHashes, receiptRequester)
+
+	rrh.addRequestedItems(unrequestedHashes, uniqueReceiptSuffix)
```

Comment:
this is more ore less duplicated with `requestByHashes` method.. why not simply reduce it to one line call:
```go
rrh.requestByHashes(rrh.shardID, receiptsHashes, factory.ReceiptTopic, uniqueReceiptSuffix)
```

Reply:
refactored